pandas/core/internals/blocks.py

```diff
@@ -1468,9 +1468,7 @@ def iget(self, col):
             elif isinstance(col, slice):
                 if col != slice(None):
                     raise NotImplementedError(col)
-                # error: Invalid index type "List[Any]" for "ExtensionArray"; expected
-                # type "Union[int, slice, ndarray]"
-                return self.values[[loc]]  # type: ignore[index]
+                return self.values[[loc]]
```

Comment:
This may not be a false positive. The EA interface does not specify (in the docstring) that a list is accepted for \_\_getitem__, although maybe we have tests in the base extension tests that enforce this. will need to check.

Reply:
Maybe the docs need to get updated, because this works:
```
>>> s=pd.array([1,2,3,4,5])
>>> s[[2,4]]
<IntegerArray>
[3, 5]
Length: 2, dtype: Int64
```

Reply:
sure for our internal EAs. it's 3rd party EAs that may not support this.

Reply:
So I guess there are two choices:

1. Require 3rd party EAs to support it and update the docs
2. Remove the `List[Any]` that I put in, and just mark the mypy error on that one place

I will assume you prefer (2), and will remove that

Reply:
will look in detail tomorrow.

Reply:
Last commits do it using (2).

Reply:
> Require 3rd party EAs to support it and update the docs

This one (not necessarily for this PR)

Reply:
One other thought on this.  There was some comment about having the EA's include a 2D signature, but I think that can be done in a subclass that implements it.  So right now, the signature for `__getitem__()` is:
```
def __getitem__(
        self, item: Union[int, slice, np.ndarray]
    ) -> Union[ExtensionArray, Any]:
```
The discussion here is (to do in another PR) to make it (and change the docs):
```
def __getitem__(
        self, item: Union[int, slice, np.ndarray, List[int]]
    ) -> Union[ExtensionArray, Any]:
```
Note the use of `List[int]` here, not `List[Any]` as I had it before.

If there is a 2D EA that wants to allow `Tuple[int, ...]`, it can have it's own signature that widens the EA signature:
```
def __getitem__(
        self, item: Union[int, slice, np.ndarray, List[int], Tuple[int, ...]]
    ) -> Union[ExtensionArray, Any]:
```
But we don't have to include that signature with `Tuple[int,...]`  in the base EA class.